projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -735,9 +740,14 @@ export class DatagridComponent<R extends B, B = any> implements OnInit, AfterVie
      *
      * If the record has a href, defaults to that. Else, defaults to index.
      */
-    @Input() trackBy: TrackByFunction<B> = (index: number, record): string => {
-        return (record as any).href;
-    };
+    @Input() set trackBy(trackByFn: TrackByFunction<B>) {
+        this._trackBy = trackByFn;
+        this.clrDgItemsTrackBy = (record) => trackByFn(undefined, record);
```

Comment:
I'm considering if we should pass a bogus `0` instead of `undefined`, to ensure we are not introducing runtime errors in the off chance someone is doing something with the `index: number`. Thoughts?

Reply:
Replaced `undefined` with `0`, filed a clarity issue and included a comment with a link to it.